src/Umbraco.Examine.Lucene/BackOfficeExamineSearcher.cs

```diff
@@ -0,0 +1,345 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Examine;
+using Umbraco.Core;
+using Umbraco.Core.Models;
+using Umbraco.Core.Models.Identity;
+using Umbraco.Core.Persistence;
+using Umbraco.Core.Services;
+using Umbraco.Web.Models.ContentEditing;
+
+namespace Umbraco.Examine
+{
+    public class BackOfficeExamineSearcher : IBackOfficeExamineSearcher
+    {
+        private readonly IExamineManager _examineManager;
+        private readonly ILocalizationService _languageService;
+        private readonly ICurrentUserAccessor _currentUserAccessor;
+        private readonly IEntityService _entityService;
+        private readonly IUmbracoTreeSearcherFields _treeSearcherFields;
+
+        public BackOfficeExamineSearcher(IExamineManager examineManager,
+           ILocalizationService languageService,
+           ICurrentUserAccessor currentUserAccessor,
+           IEntityService entityService,
+           IUmbracoTreeSearcherFields treeSearcherFields)
+        {
+            _examineManager = examineManager;
+            _languageService = languageService;
+            _currentUserAccessor = currentUserAccessor;
+            _entityService = entityService;
+            _treeSearcherFields = treeSearcherFields;
+        }
+
+        public IEnumerable<ISearchResult> Search(string query, UmbracoEntityTypes entityType, int pageSize, long pageIndex, out long totalFound, string searchFrom = null, bool ignoreUserStartNodes = false)
+        {
+            var sb = new StringBuilder();
+
+            string type;
+            var indexName = Constants.UmbracoIndexes.InternalIndexName;
+            var fields = _treeSearcherFields.GetBackOfficeFields().ToList();
+
+            // TODO: WE should try to allow passing in a lucene raw query, however we will still need to do some manual string
```

Comment:
I know this is copied, but can we remove this ?

Reply:
yep, we can defo remove